config/string_list.go

```diff
@@ -0,0 +1,76 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+)
+
+// StringList represents the "poor man's list" that terraform uses
+// internally
+type StringList string
+
+// This is the delimiter used to recognize and split StringLists
+//
+// It plays two semantic roles:
+//   * It introduces a list
+//   * It terminates each element
+//
+// Example representations:
+// []             => SLD
+// [""]           => SLDSLD
+// [" "]          => SLD SLD
+// ["foo"]        => SLDfooSLD
+// ["foo", "bar"] => SLDfooSLDbarSLD
+// ["", ""]       => SLDSLDSLD
+const stringListDelim = `B780FFEC-B661-4EB8-9236-A01737AD98B6`
+
+// Build a StringList from a slice
+func NewStringList(parts []string) StringList {
+	// We have to special case the empty list representation
+	if len(parts) == 0 {
+		return StringList(stringListDelim)
+	}
+	return StringList(fmt.Sprintf("%s%s%s",
+		stringListDelim,
+		strings.Join(parts, stringListDelim),
+		stringListDelim,
+	))
+}
+
+// Returns an element at the index, wrapping around the length of the string
+// when index > list length
+func (sl StringList) Element(index int) string {
+	if sl.Length() == 0 {
+		return ""
+	}
+	return sl.Slice()[index%sl.Length()]
+}
+
+// Returns the length of the StringList
+func (sl StringList) Length() int {
+	return len(sl.Slice())
+}
+
+// Returns a slice of strings as represented by this StringList
+func (sl StringList) Slice() []string {
+	parts := strings.Split(string(sl), stringListDelim)
+
+	switch len(parts) {
+	case 0, 1:
+		return []string{}
+	case 2:
+		return []string{""}
+	}
+
+	// strip empty elements generated by leading and trailing delimiters
+	return parts[1 : len(parts)-1]
+}
+
+func (sl StringList) String() string {
+	return string(sl)
+}
```

Comment:
Decided to go with an explicit function to "convert" to a string. Felt right but would appreciate commentary on the most go-like thing to do here.